src/helpers/index.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Allowed implicit type dependencies,
+ * given that a project may not choose to use this helper,
+ * *or* uses Vue + Vuex without Webpack at all
+ */
+/* tslint:disable:no-implicit-dependencies */
+import { MinifyOptions } from "terser";
+import { Configuration } from "webpack";
+
+export function preserveFunctionNamesWithTerser(config: Configuration) {
+    try {
+        if (process.env.NODE_ENV === "production") {
```

Comment:
Perhaps a silly question, but are we guaranteeing that the production node environment is called "production" in all cases and not "Production" or "PRODUCTION" or "prod" ?

Reply:
Not a silly question at all. We're assuming that this helper is used in the context of a Vue CLI 3 project, in which case that will be the environment name for `NODE_ENV`.

In the future, the helper could be modified to *create* these configuration options, if they don't yet exist. At which point it could be used outside the context of Vue CLI, and in any Webpack project. When we'd need to allow a consumer to manually specify as an arg "this is when I want to add the config options"

For now, it's a safe assumption, until someone would like to use the helper outside of Vue CLI

Reply:
https://cli.vuejs.org/guide/mode-and-env.html#modes